src/SprykerEco/Yves/Adyen/Plugin/Payment/CreditCardPaymentMapperPlugin.php

```diff
@@ -26,12 +26,24 @@ public function setPaymentDataToQuote(QuoteTransfer $quoteTransfer, Request $req
             return;
         }
 
-        $quoteTransfer
+        $encryptedCardNumber = $request->get(AdyenApiRequestConfig::ENCRYPTED_CARD_NUMBER_FIELD);
+        $encryptedExpiryMonth = $request->get(AdyenApiRequestConfig::ENCRYPTED_EXPIRY_MONTH_FIELD);
+        $encryptedExpiryYear = $request->get(AdyenApiRequestConfig::ENCRYPTED_EXPIRY_YEAR_FIELD);
+        $encryptedSecurityCode = $request->get(AdyenApiRequestConfig::ENCRYPTED_SECURITY_CODE_FIELD);
+
+        if ($encryptedCardNumber === null && $encryptedExpiryMonth === null
+            && $encryptedExpiryYear === null && $encryptedSecurityCode === null) {
+            return;
+        }
```

Comment:
This condition looks wrong as for me. What if only card number is missing in request?

Reply:
Refactored.